mofa/simulation/graspa_sycl.py

```diff
@@ -0,0 +1,298 @@
+
+        Args:
+            name (str): Name of the MOF (excluding the ".cif" extension).
+            cp2k_path (str): Path to the Chargemol output file.
+            adsorbate (str): Adsorbate name. Supported values: 'H2', 'CO2', 'N2'.
+            temperature (float): Simulation temperature in Kelvin (K).
+            pressure (float): Simulation pressure in Pascal (Pa).
+            cutoff (float, optional): Van der Waals and Coulomb cutoff distance in Angstroms. Defaults to 12.8 Å.
+            n_cycle (int, optional): Number of initialization and production cycles. Defaults to 500.
+            chargemol_fname (str, optional): Filename of the Chargemol output. Defaults to "DDEC6_even_tempered_net_atomic_charges.xyz".
+
+        Returns:
+            tuple[float, float, float, float]: Computed uptake (U) and error (E) from gRASPA-sycl in the following order:
+                - U (mol/kg)
+                - E (mol/kg)
+                - U (g/L)
+                - E (g/L)
+        """
+
+        out_dir = (
+            self.run_dir / f"{name}_{adsorbate}_{temperature}_{pressure:0e}"
+        )
+        out_dir.mkdir(parents=True, exist_ok=True)
+
+        atoms = self._get_cif_from_chargemol(
+            cp2k_path, chargemol_fname=chargemol_fname
+        )
+        # Write CIF file with charges
+        self._write_cif(atoms, out_dir=out_dir, name=name + ".cif")
+
+        # Copy other input files (simulation.input, force fields and definition files) from template folder.
+        subprocess.run(f"cp {_file_dir}/* {out_dir}/", shell=True)
+
+        [uc_x, uc_y, uc_z] = self._calculate_cell_size(atoms=atoms)
+
+        # Modify input from template simulation.input
+        with (
+            open(f"{out_dir}/simulation.input", "r") as f_in,
+            open(f"{out_dir}/simulation.input.tmp", "w") as f_out,
+        ):
+            for line in f_in:
+                if "NCYCLE" in line:
+                    line = line.replace("NCYCLE", str(n_cycle))
+                if "ADSORBATE" in line:
+                    line = line.replace("ADSORBATE", adsorbate)
+                if "TEMPERATURE" in line:
+                    line = line.replace("TEMPERATURE", str(temperature))
+                if "PRESSURE" in line:
+                    line = line.replace("PRESSURE", str(pressure))
+                if "UC_X UC_Y UC_Z" in line:
+                    line = line.replace(
+                        "UC_X UC_Y UC_Z", f"{uc_x} {uc_y} {uc_z}"
+                    )
+                if "CUTOFF" in line:
+                    line = line.replace("CUTOFF", str(cutoff))
+                if "CIFFILE" in line:
+                    line = line.replace("CIFFILE", name)
+                f_out.write(line)
+
+        shutil.move(
+            f"{out_dir}/simulation.input.tmp", f"{out_dir}/simulation.input"
+        )
+
+        # Run gRASPA-sycl
+        subprocess.run(self.graspa_sycl_command, shell=True, cwd=out_dir)
```

Comment:
Rather than using ">>" in the shell command to write stdout to a file, you can have Python direct the outputs. There's probably not much of a performance difference, but using the Python method does mean a user forgetting to put ">>" into the command doesn't break everything.

An example of Python directing: https://github.com/globus-labs/mof-generation-at-scale/blob/main/mofa/simulation/lammps.py#L160